library/MiscUtils.cpp

```diff
@@ -321,7 +321,45 @@ static uint16_t character_table[256] = {
     0xB0,   0x2219, 0xB7,   0x221A, 0x207F, 0xB2,   0x25A0, 0xA0
 };
 
-std::string DF2UTF(const std::string &in)
+#ifdef WIN32
+std::string UTF16ToUTF8(const std::wstring &in)
+{
+    // get buffer size in UTF-8
+    int size = WideCharToMultiByte(CP_UTF8, 0, &in[0], in.size(), nullptr, 0, nullptr, nullptr);
+
+    // convert to active codepage
+    std::string out(unsigned(size), 0);
+    WideCharToMultiByte(CP_UTF8, 0, &in[0], in.size(), &out[0], out.size(), nullptr, nullptr);
+
+    return out;
+}
+
+std::wstring UTF8ToUTF16(const std::string &in)
+{
+    // get buffer length
+    int size = MultiByteToWideChar(CP_UTF8, 0, &in[0], in.size(), nullptr, 0);
+
+    // convert to active codepage
+    std::wstring out(unsigned(size), 0);
+    MultiByteToWideChar(CP_UTF8, 0, &in[0], in.size(), &out[0], out.size());
+
+    return out;
+}
+
+std::string UTF16ToCurrentCodePage(const std::wstring &in)
+{
+    // get buffer size in current codepage
+    int size = WideCharToMultiByte(CP_THREAD_ACP, 0, &in[0], in.size(), nullptr, 0, nullptr, nullptr);
```

Comment:
CP_THREAD_ACP isn't actually the correct codepage - it's "ANSI", while console windows typically use "OEM"